tests/test_diffing/test_stack_differ.py

```diff
@@ -332,6 +332,17 @@ def test_diff__deployed_stack_has_default_values__passes_different_value__compar
             self.expected_deployed_config, self.expected_generated_config
         )
 
+    def test_diff__deployed_stack_has_default_values_one_of_them_none__compares_different_configs(
+        self,
+    ):
+        self.deployed_parameters["new"] = "default value"
+        self.deployed_parameter_defaults["new"] = None
+        self.parameters_on_stack_config["new"] = ""
+        self.differ.diff(self.actions)
+        self.command_capturer.compare_stack_configurations.assert_called_with(
+            self.expected_deployed_config, self.expected_generated_config
+        )
+
```

Comment:
@jfalkenstein this test passes with or without my patch so it's evidently not a valid test. If you can see what's wrong that will be most appreciated!

Reply:
Ok, so the lines of code you need to look at are here: https://github.com/Sceptre/sceptre/blob/master/sceptre/diffing/stack_differ.py#L257

Basically, the way the diffing works _right now_ is:
* If a given parameter is defined on the stack config, diff it's value with the deployed config
* If a given parameter is NOT defined on the stack config and the deployed parameter value is identical to its default value, delete the parameter from the deployed config, since we're relying on the default value so there should be no diff detected.

However, the change I think you'll need to make is to add a new condition to that:
* If a parameter is defined on the stack config as `None` and the parameter has a default value, delete that parameter off BOTH the generated config and the deployed config.

Regarding this test, then, I think part of your problem is that you're saying that the default value on the CloudFormation stack for the "new" parameter is `None`, but that's not a valid default value. What you need to test is:

First, you should update the expected_generated_config and expected_deployed_config properties so that the parameters are deepcopied as well. That is important. 

```python
        # The default value for the "new" parameter is "default_value"
        self.deployed_parameter_defaults["new"] = "default_value"
        # It just so happens that the deployed parameter value IS the default value, 
        # meaning we didn't override it
        self.deployed_parameters["new"] = "default value"
        # On the Stack Config, we're setting the value to None, which means we're effectively 
        # NOT setting any value at all.
        self.parameters_on_stack_config["new"] = None
        # We need to modify the expected generated config to be diffed. Look at the property 
        # for more details on how it's created to start with
        expected_generated_config = self.expected_generated_config
        expected_deployed_config = self.expected_deployed_config
        # We should expect that there is no "new" parameter specified at all in the generated_config or the 
        # deployed_config        
        del expected_generated_config.parameters['new']
        del expected_deployed_config.parameters['new']

        self.differ.diff(self.actions)
        self.command_capturer.compare_stack_configurations.assert_called_with(
            self.expected_deployed_config, self.expected_generated_config
        )
```

You'll also need test cases for when:
* There is no default value on the parameter
* There is no deployed config
* When the parameter value on the deployed config is NOT the default value but the parameter on the generated config is None (in which case it defaults to the deployed config).